core-container/entry.sh

```diff
@@ -23,145 +23,133 @@ function echoerr() {
+function wait_kcp_ready() {
+    echo "Waiting for kcp to be ready... this may take a while."
     (
-        until [ "$(kubectl logs $(kubectl get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c space-manager | grep '***READY***')" != "" ]; do
-            sleep 10
+        until [ "$(kubectl --kubeconfig $host_kubeconfig logs $(kubectl --kubeconfig $host_kubeconfig get pod --selector=app=kubestellar -o jsonpath='{.items[0].metadata.name}') -c kcp | grep '***READY***')" != "" ]; do
+           sleep 10
         done
     )
+}
 
+function get_kcp_kubeconfig() {
+    wait_kcp_ready
+    while [ $(KUBECONFIG=$host_kubeconfig kubectl get secret kubestellar | grep -c -e kubestellar) != "1" ]; do 
+        echo "Waiting for kubestellar secret."
+        sleep 1
+    done
+    kcp_kubeconfig_dir=/home/kubestellar/.kcp
+    kcp_kubeconfig=${kcp_kubeconfig_dir}/admin.kubeconfig
+    echo "Copying the admin.kubeconfig from kubestellar seret..."
+    mkdir -p $kcp_kubeconfig_dir
+    kubectl --kubeconfig $host_kubeconfig get secrets kubestellar -o jsonpath='{.data.admin\.kubeconfig}' | base64 --decode > $kcp_kubeconfig
+}
+
+function create_kcp_provider() {
+    get_kcp_kubeconfig     # kcp is created in a seperate container 
     echo "Delete the kcp provider secret if it already exists, and then create it."
-    if ! kubectl delete secret -n ${NAMESPACE} kcpsec ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete secret -n ${NAMESPACE} kcpsec ; then
         echo "Nothing to delete."
     fi
-    kubectl get secrets kubestellar -o 'go-template={{index .data "admin.kubeconfig"}}' | base64 --decode > kcpsecret
-    kubectl create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig="kcpsecret"    
-    rm kcpsecret
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create secret generic -n ${NAMESPACE} kcpsec --from-file=kubeconfig=$kcp_kubeconfig   
 
     echo "Delete the kcp provider object if it already exists, and then create it."
-    if ! kubectl delete spaceproviderdesc pkcp ; then
+    if ! kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG delete spaceproviderdesc $PROVIDER_NAME ; then
         echo "Nothing to delete."
     fi
-    kubectl create -f - <<EOF
+    kubectl --kubeconfig $SPACE_MANAGER_KUBECONFIG create -f - <<EOF
 apiVersion: space.kubestellar.io/v1alpha1
 kind: SpaceProviderDesc
 metadata:
-  name: pkcp
+  name: $PROVIDER_NAME
 spec:
   ProviderType: "kcp"
   SpacePrefixForDiscovery: "ks-"
   secretRef:
     namespace: ${NAMESPACE}
     name: kcpsec
 EOF
-
-    if [ "$SPACE_PROVIDER" == "kubeflex" ]; then
-        echo "Waiting for the kubeflex provider to be ready... this may take a while."
-        (
-            until [ "$(kubectl get pods -A | grep kubeflex-controller-manager | grep Running)" != "" ]; do
-                sleep 10
-            done
-        )
+    echo "Waiting for default spaceprovider to reach the Ready phase."
```

Comment:
Should "default" be a fixed literal here?

Reply:
removed default.